clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -930,7 +932,16 @@ invariant_checker_t::check_for_pc_discontinuity(
     std::string error_msg = "";
     bool have_branch_target = false;
     addr_t branch_target = 0;
-    const addr_t prev_instr_trace_pc = prev_instr.instr.addr;
+#ifdef UNIX
+    // This needs to be last_instr_in_cur_context_ because we want to check continuity
+    // with the last instruction at the same signal nesting depth as the current instr
+    // instead of the previous executed instr (which may be from an inner nested signal if
+    // a signal just ended).
+    addr_t prev_instr_trace_pc = shard->last_instr_in_cur_context_.instr.addr;
+#else
+    addr_t prev_instr_trace_pc = prev_instr.instr.addr;
+#endif
+
```

Comment:
What about the 4 different i#5912 comments in this file: I assumed they would be resolved by this PR.  E.g., the prev_instr_decode_ cache: I thought that was broken for for the signal checks being added here.